services/command-router/src/main/java/org/eclipse/hono/commandrouter/impl/kafka/KafkaBasedCommandConsumerFactoryImpl.java

```diff
@@ -241,6 +247,18 @@ public Future<Void> start() {
         return CompositeFuture.all(commandHandler.start(), kafkaConsumer.start()).mapEmpty();
     }
 
+    private void registerTenantCreationListener() {
+        NotificationEventBusSupport.registerConsumer(vertx, TenantChangeNotification.TYPE,
+                notification -> {
+                    if (lifecycleStatus.isStarted() && LifecycleChange.CREATE.equals(notification.getChange())
```

Comment:
LifecycleChange.CREATE == notification.getChange() ?

my understanding is that if *lifecycleStatus.isStarted()* returns `false` then the KafkaConsumer will create the topic and subscription *on-the-fly* when an adapter first registers a command handler for the tenant, meaning that this is just an optimization, right? If so, then IMHO we should add a corresponding comment here so that readers do not think that we missed out on a relevant case ...

Reply:
This is an optimization, yes. I've added a corresponding comment.

> my understanding is that if _lifecycleStatus.isStarted()_ returns `false` then the KafkaConsumer will create the topic and subscription _on-the-fly_ when an adapter first registers a command handler for the tenant, 

I guess you mean _lifecycleStatus.isStarted()_ being `true` here? That is the condition for `ensureTopicIsAmongSubscribedTopicPatternTopics` - the topic creation and consumer-rebalance trigger in that method don't make sense when the consumer isn't ready yet or if the consumer is stopping/stopped.

Reply:
my point is that the notification will be handled here only if *lifecycleStatus.isStarted()* returns `true` here, i.e. the notification will not be handled if it returns `false`. However, the topic for the tenant needs to be created at some point in time, right?

Reply:
Yes, if _this_ consumer isn't _started_, then topic creation will either be triggered by another consumer instance (possibly in another Command Router instance), or as part of the `createCommandConsumer` invocation.